contrib/hash_ring.c

```diff
@@ -172,35 +168,45 @@ int hash_ring_add_items(hash_ring_t *ring, hash_ring_node_t *node) {
     uint64_t keyInt;
 
     // Resize the items array
-    void *resized = realloc(ring->items, (sizeof(hash_ring_item_t*) * ring->numNodes * ring->numReplicas));
+    void *resized = realloc(ring->items, (sizeof(hash_ring_item_t*) * (ring->numItems + node->numReplicas)));
```

Comment:
This seems wrong? It's resizing a buffer on ring based on the value of a property on node. What if I call hash_ring_add_items with a node with a higher number of replicas (let's say 2), and then call hash_ring_add_items with a node with a lower number of replicas (let's say 1).

Reply:
Just kidding, this is fine. (ring->numItems is incremeted by node->numReplicas later in this function).